ddl/index.go

```diff
@@ -52,6 +52,10 @@ func buildIndexInfo(tblInfo *model.TableInfo, unique bool, indexName model.CIStr
 			return nil, errors.Trace(errIncorrectPrefixKey)
 		}
 
+		if ic.Length > 255 {
```

Comment:
"For example, a prefix can be up to 767 bytes long for InnoDB tables or 3072 bytes if the innodb_large_prefix option is enabled."
See: http://dev.mysql.com/doc/refman/5.7/en/create-index.html

We can use the constraint of InnoDB. And it is better to use a const.